src/ivalue_manager.rs

```diff
@@ -637,23 +637,56 @@ impl<'a> Manager for RedisIValueJsonKeyManager<'a> {
         }
     }
 
+    ///
+    /// following https://github.com/Diggsey/ijson/issues/23#issuecomment-1377270111
+    ///
     fn get_memory(&self, v: &Self::V) -> Result<usize, RedisError> {
         let res = size_of::<IValue>()
             + match v.type_() {
-                ValueType::Null | ValueType::Bool | ValueType::Number => 0,
+                ValueType::Null | ValueType::Bool => 0,
+                ValueType::Number => {
+                    let num = v.as_number().unwrap();
+                    if num.has_decimal_point() {
+                        // 64bit float
+                        16
+                    } else if num > &INumber::from(-128) && num <= &INumber::from(383) {
+                        // 8bit
+                        0
+                    } else if num > &INumber::from(-8_388_608) && num <= &INumber::from(8_388_607) {
+                        // 24bit
+                        4
+                    } else {
+                        // 64bit
+                        16
+                    }
+                }
                 ValueType::String => v.as_string().unwrap().len(),
```

Comment:
Be aware that this may signficantly over-estimate the size, since strings are interned.

Reply:
yes, I'm aware but it still give a single JSON it's potential cost